### PR TITLE
[Dashboard][Recruitment] Added candidate age ranges pie chart to recruitment dashboard

### LINKED ISSUE #8133: [Dashboard / Recruitment by Age] To create a graph visualizing the recruitment by Age.
**Is your feature request related to an existing module?**
dashboard --> Recruitment 

**Describe the solution you'd like**
As an part of ongoing discussions with some projects the idea of having a chart showing the **Recruitment by Ag**e was proposed (specifically CTU and CCNA).  

**Describe alternatives you've considered**
In a first approach the age ranges considered will be as follow: 
5 year gap:  less than 60, 60-64, 65-69, 70 - 74, 75-79, 80-85, 86 or more (which is the requirement for this specific projects) 
In a second phase of this task the age ranges will be configurable on the **config module.** 

**Additional context**
Requested by CTU and CCNA.

### PR DESCRIPTION
**Pull request by satvik-tha-god**
## Brief summary of changes

* Added the candidate age ranges Pie chart for the following ranges:  less than 60, 60-64, 65-69, 70 - 74, 75-79, 80-85, 86 or more to recruitment
* Updated the layout of the other components to fit better
* Added the necessary AJAX and utility JS functions
* Added PHP backend code for related

## Screenshots:
Screenshot from 2022-07-11 19-42-36
Screenshot from 2022-07-11 19-44-14

#### Link(s) to related issue(s)

* Resolves #8133   (Reference the issue this fixes, if any.)

### CODE REVIEW

**Review by racostas — COMMENTED** [review]
Hi Satvik, nice work !! For making this graph more generalizable let's please take the minimum and maximum age from the project configuration.

Also let's try to show in the graph not just the percentage of candidates in the range but also the "total count". This maybe will have to be done in a new PR.

**Comment by christinerogers** [comment]
@racostas is this ready for @driusan to review?

**Comment by christinerogers** [comment]
@racostas probably best if you re-review

**Review by racostas — APPROVED** [review]
LGTM.

File: modules/statistics/php/charts.class.inc [hunk 1]
```diff
@@ -129,6 +131,93 @@ class Charts extends \NDB_Page
...
+     * @return ResponseInterface
+     */
+    private function _handleAgePieData()
+    {
+        $DB     = \NDB_Factory::singleton()->database();
+        $config = \NDB_Config::singleton();
+        $Dates  =$DB->pselect(
+            "SELECT c.DoB, c.Date_registered 
+                           FROM candidate c
+                           WHERE c.DoB IS NOT NULL
```
Comment (driusan) [inline modules/statistics/php/charts.class.inc:146]: this is missing `c.Active='Y'`

**Comment by racostas** [comment]
> see last two comments about console.error and fixing the query

^ @satvik-tha-god

**Review by racostas — CHANGES_REQUESTED** [review]
Hi @satvik-tha-god , it seems the test suit is failing after the latest changes. Could you please take a look when you get a chance ? 
Many thanks.

**Comment by racostas** [comment]
A wider/extensible solution is currently given to this issue at CCNA project. The plan is to merge all this changes from CCNA to LORIS core soon. For this reason I think this PRs got outdated. It worth to mention nevertheless that the principles and solutions prosed here were the base for the solution currently been implemented at CCNA (the one that will be merged back to LORIS core). 

For this reason I'm closing this PR.

**Pull request closed without merging**